sdk/metric/pipeline.go

```diff
@@ -134,3 +140,189 @@ func (p *pipeline) produce(ctx context.Context) (metricdata.ResourceMetrics, err
+}
+
+func (m *multierror) errorOrNil() error {
+	if len(m.errors) == 0 {
+		return nil
+	}
+	return fmt.Errorf(strings.Join(m.errors, "; "))
+}
+
+func (m *multierror) append(err error) {
+	m.errors = append(m.errors, err.Error())
+}
+
+// instrumentIdentifier is used to identify multiple instruments being mapped to the same aggregator.
+// e.g. using an exact match view with a name=* view.
+// You can't use a view.Instrument here because not all Aggregators are hashable.
+type instrumentIdentifier struct {
+	scope       instrumentation.Scope
+	name        string
+	description string
+}
+
+func createAggregators[N int64 | float64](rdr Reader, views []view.View, inst view.Instrument) (map[instrumentIdentifier]internal.Aggregator[N], error) {
+	aggs := map[instrumentIdentifier]internal.Aggregator[N]{}
+	errs := &multierror{}
+	for _, v := range views {
+		inst, match := v.TransformInstrument(inst)
+
+		ident := instrumentIdentifier{
+			scope:       inst.Scope,
+			name:        inst.Name,
+			description: inst.Description,
+		}
+
+		if _, ok := aggs[ident]; ok || !match {
+			continue
+		}
+
+		if inst.Aggregation == nil {
+			inst.Aggregation = rdr.aggregation(inst.Kind)
+		} else if _, ok := inst.Aggregation.(aggregation.Default); ok {
+			inst.Aggregation = rdr.aggregation(inst.Kind)
+		}
+
+		if err := isAggregatorCompatible(inst.Kind, inst.Aggregation); err != nil {
+			global.Error(err, "creating aggregator", "instrumentKind", inst.Kind, "aggregation", inst.Aggregation)
+			errs.append(err)
+			continue
+		}
+
+		agg := createAggregator[N](inst.Aggregation, rdr.temporality(inst.Kind), isMonotonic(inst.Kind))
+		if agg != nil {
+			// TODO (#3011): If filtering is done at the instrument level add here.
+			// This is where the aggregator and the view are both in scope.
+			aggs[ident] = agg
+		}
+	}
+	return aggs, errs.errorOrNil()
+}
+
+func isMonotonic(kind view.InstrumentKind) bool {
+	switch kind {
+	case view.AsyncCounter, view.SyncCounter, view.SyncHistogram:
+		return true
+	}
+	return false
+}
+
+// createAggregator takes the config (Aggregation and Temporality) and produces a memory backed Aggregator.
+// TODO (#3011): If filterting is done by the Aggregator it should be passed here.
+func createAggregator[N int64 | float64](agg aggregation.Aggregation, temporality metricdata.Temporality, monotonic bool) internal.Aggregator[N] {
+	switch agg := agg.(type) {
+	case aggregation.Drop:
+		return nil
+	case aggregation.LastValue:
+		return internal.NewLastValue[N]()
+	case aggregation.Sum:
+		if temporality == metricdata.CumulativeTemporality {
+			return internal.NewCumulativeSum[N](monotonic)
+		}
+		return internal.NewDeltaSum[N](monotonic)
+	case aggregation.ExplicitBucketHistogram:
+		if temporality == metricdata.CumulativeTemporality {
+			return internal.NewCumulativeHistogram[N](agg)
+		}
+		return internal.NewDeltaHistogram[N](agg)
+	}
+	return nil
+}
+
+var errIncompatibleAggregation = errors.New("incompatible aggregation")
```

Comment:
Thank you. I filed this tracking issue. https://github.com/open-telemetry/opentelemetry-specification/issues/2710